bioptim/dynamics/integrator.py

```diff
@@ -719,9 +684,6 @@ def dxdt(
             else:
                 raise ValueError("Unknown defects type. Please use 'explicit' or 'implicit'")
 
-            # Add contribution to the end state
-            states_end += self._d[j] * states[j + 1]
```

Comment:
It seems suspicious that this line was removed

Reply:
I put it out of the loop and I entirely interpolate at 1.0 (i.e. the end of the interval) before.